lib/autorestPlugin/pluginHost.ts

```diff
@@ -140,7 +160,7 @@ export async function openApiValidationExample(
           for (const responseCode of utils.getKeys(scenarioItem.responses)) {
```

Comment:
>     for (const responseCode of utils.getKeys(scenarioItem.responses)) { [](start = 5, length = 72)

maybe we can replace these nested for's wiht reducers ?
```
utils.getKeys(innerErrors).reduce( (acc, error) => {}, formattedResult); 
```

Reply:
same for other cases

---
In reply to: [194471819](https://github.com/Azure/oav/pull/257#discussion_r194471819) [](ancestors = 194471819)

Reply:
I agree that we need to use less `for` cycles. As far as I see, we don't need to accumulate results, so `map` + `filter` may work better here.

Reply:
don't we accumulate the results in `formattedResult` ? but yea if we don't accumulate anything, `map` for sure